rpm/bundle-rust-npm.py

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/python3
+# --- BEGIN COPYRIGHT BLOCK ---
+# Copyright (C) 2024 Red Hat, Inc.
+# All rights reserved.
+#
+# License: GPL (version 3 or any later version).
+# See LICENSE for details.
+# --- END COPYRIGHT BLOCK ---
+#
+# PYTHON_ARGCOMPLETE_OK
+
+import os
+import sys
+import subprocess
+import time
+import signal
+import argparse
+import argcomplete
+import shutil
+import json
+import re
+from typing import List, Dict, Tuple, Set
+from lib389.cli_base import setup_script_logger
+from lib389.utils import ensure_list_str, ensure_str
+
+SPECFILE_COMMENT_LINE = 'Bundled cargo crates list'
+START_LINE = f"##### {SPECFILE_COMMENT_LINE} - START #####\n"
+END_LINE = f"##### {SPECFILE_COMMENT_LINE} - END #####\n"
+
+IGNORED_RUST_PACKAGES: Set[str] = {"librslapd", "librnsslapd", "slapd", "slapi_r_plugin", "entryuuid", "entryuuid_syntax", "pwdchan"}
+IGNORED_NPM_PACKAGES: Set[str] = {"389-console"}
+PACKAGE_REGEX = re.compile(r"(.*)@(.*)")
+
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    description="""Add 'Provides: bundled(crate(foo)) = version' 'Provides: bundled(npm(bar)) = version' to a Fedora based specfile.
+Additionally, add a helper comment with a comulated License metainfo which is based on Cargo.lock and Package-lock.json files content.
+You need to have 'cargo install cargo-license' and 'npm install -g license-checker' to be able to run this script.""")
```

Comment:
I'd prefer not install NPM packages globally. I usually run these through `npx`.